server.go

```diff
@@ -467,7 +467,16 @@ func (s *Server) inheritClientSession(pk packets.Packet, cl *Client) bool {
 			return false // [MQTT-3.2.2-3]
 		}
 
-		cl.State.Inflight = existing.State.Inflight // [MQTT-3.1.2-5]
+		infLen := existing.State.Inflight.Len()
```

Comment:
This is contrary to the mqtt v5 spec:
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Flow_Control

> 4.9 Flow Control [...] The send quota and Receive Maximum value are not preserved across Network Connections, and are re-initialized with each new Network Connection as described above. They are not part of the session state.

and 

> 3.1.2.11.3 Receive Maximum [...] The value of Receive Maximum applies only to the current Network Connection.

Reply:
`cl.State.Inflight = existing.State.Inflight`

This code overwrites the quota values of the client to zero, causing an ErrReceiveMaximum when the client publishes the message.

`  
func (s *Server) processPublish(cl *Client, pk packets.Packet) error {
	if !cl.Net.Inline && !IsValidFilter(pk.TopicName, true) {
		return nil
	}

	if atomic.LoadInt32(&cl.State.Inflight.receiveQuota) == 0 {
		return s.DisconnectClient(cl, packets.ErrReceiveMaximum) // ~[MQTT-3.3.4-7] ~[MQTT-3.3.4-8]
	}
`